src/common/state/redis.c

```diff
@@ -153,22 +163,34 @@ db_handle *db_connect(const char *address,
     freeReplyObject(reply);
     reply = redisCommand(context, "MULTI");
     freeReplyObject(reply);
-    reply = redisCommand(
-        context,
-        "HMSET db_clients:%b client_type %s address %s:%d db_client_id %b",
-        (char *) client.id, sizeof(client.id), client_type, client_addr,
-        client_port, (char *) client.id, sizeof(client.id));
+    reply = redisCommand(context,
+                         "HMSET db_clients:%b client_type %s address %s:%d "
+                         "db_client_id %b aux_address %s",
+                         (char *) client.id, sizeof(client.id), client_type,
+                         client_addr, client_port, (char *) client.id,
+                         sizeof(client.id), aux_address);
+    CHECK_REDIS_CONNECT(redisContext, context,
+                        "redis_db_connect failed on HMSET with aux_address");
+    CHECKM(reply != NULL, "db_connect failed on HMSET");
+    /* allocate space for the published message */
+    char *tmpbuf =
+        malloc(strlen(client_type) + 1 + strlen(aux_address + 1 + 1));
```

Comment:
why the + 1 + 1 inside of the strlen argument?

If I think about it, why not get rid of tmpbuf altogether and do all the formatting in the redisCommand like

"%b:%s %s"

?